LSP-rust-analyzer.sublime-settings

```diff
@@ -1,10 +1,12 @@
 {
 	"selector": "source.rust",
-	"command": ["${storage_path}/LSP-rust-analyzer/rust-analyzer"],
+	"command": ["${rust_analyzer_binary_path}"],
```

Comment:
Why don't the user just set the path to rust analyser binary path itself?

```
"command": ["path/to/rust_analyzer_binary"],
```

Reply:
RE: because that would still cause this popup https://github.com/sublimelsp/LSP-rust-analyzer/issues/68#issuecomment-1126961835

Reply:
@parcimonic do you consider this feature useful for yourself or was it just to fix the issue with downloading managed binary?

As you've pointed out yourself, using system binary can result in settings not matching and server complaining (but probably still working fine).

Not sure what would be a typical use case for using it this way? Maybe just to bump the server version before this package gets around doing that?

Reply:
@rchl I've created the PR because of personal preferences, mainly avoiding programs downloading binaries on my OS without my consent.

I've referenced issue #68 because, in my opinion, the PR does fulfill the request in the title: "Users should be able to bring their own rust-analyzer binary" (unrelated to the issue itself, as noted by @predragnikolic) and this comment: "But yeah I'll work on a way to disable auto downloading and allow users to vendor their own version"

Maybe I've done something wrong, but in all my tests before working on the code changes, the LSP-rust-analyzer package *always* downloaded and used its own version of the rust-analyzer binary instead of using the one I've installed on my OS.

Additionally, I have no objections to having this PR being closed and not merged, if this is the maintainers decision.

Reply:
I also prefer using a binary that I've installed myself (through my OS' package manager, for example) rather than using the auto-downloaded binary. I do understand that the experience may differ from the standard due to settings/versions/etc, but that is up to the end user (in this case, myself) to deal with it.
+1 for this PR.

Reply:
LSP provides a way that enables users to point to a local rust_analyzer_binary.

Users can add their own client configuration to LSP.sublime-settings (for more info see [this](https://lsp.sublimetext.io/guides/client_configuration/)),
but they have to uninstall LSP-rust-analyzer.
```jsonc
// LSP.sublime-settings
{
	"clients": {
		"rust-analyzer": {
			"enabled": true,
			"command": [
				"C:\\Users\\Predrag\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\bin\\rust-analyzer.exe"
			],
			"selector": "source.rust",
		},
	}
}
```

Maybe people weren't aware of this functionality and thus this PR?

----
My summary.

- This PR tries to add something that is already possible.

- Users could point to an older version of rust-analyzer that could potentially be incompatible with the code in LSP-rust-analyzer.

> I do understand that the experience may differ from the standard due to settings/versions/etc, but that is up to the end user (in this case, myself) to deal with it.

Like I said if the user want to configure they can with client configuration.

One of the point of LSP-* plugins is to automate things for user, 
so they do not have to (miss)configure options. :)

Reply:
Hi @predragnikolic , you are right, and in fact that is the way I use rust-analyzer with Sublime LSP.

When I worked on this PR (~2 months ago), even though you could use rust-analyzer this way, you'd be missing some features from LSP-rust-analyzer (e.g. inlay hints, additional sublime actions).

Nowadays, at least the inlay hints have been added to the base Sublime LSP package (following LSP standardization of it), so it doesn't matter too much which way the user chooses to use rust-analyzer.

I hope this makes it clear. I'll proceed and close this PR, since the gap between the two methods (OS-provided, LSP-rust-analyzer package) is much smaller now.